🐛 Tried to add a Frame Processor Plugin with a name that already exists!

### What's happening?

I wanted to create a Frame Plugin on Android following guide line but I get always the error in the title

![image](https://github.com/mrousavy/react-native-vision-camera/assets/11539474/b5e52f52-a8f0-4b89-823b-2b1f3c32dfc6)

**XyzFrameProcessorPlugin.java**
```
package com.giuseppe.ravida.exponativeexamplev331;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.mrousavy.camera.frameprocessor.Frame;
import com.mrousavy.camera.frameprocessor.FrameProcessorPlugin;

import java.util.Map;

public class XyzFrameProcessorPlugin extends FrameProcessorPlugin {
  @Nullable
  @Override
  public Object callback(@NonNull Frame frame, @Nullable Map<String, Object> arguments) {
    // code goes here
    return "Hello World";
  }
}

```

**XyzFrameProcessorPluginPackage.java**
```
package com.giuseppe.ravida.exponativeexamplev331;

import androidx.annotation.NonNull;
import com.facebook.react.ReactPackage;
import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.uimanager.ViewManager;
import com.mrousavy.camera.frameprocessor.FrameProcessorPlugin;
import com.mrousavy.camera.frameprocessor.FrameProcessorPluginRegistry;

import java.util.*;

public class XyzFrameProcessorPluginPackage implements ReactPackage {
  XyzFrameProcessorPluginPackage() {
    FrameProcessorPluginRegistry.addFrameProcessorPlugin("xyz", options -> new XyzFrameProcessorPlugin());
  }

  @NonNull
  @Override
  public List<NativeModule> createNativeModules(@NonNull ReactApplicationContext reactContext) {
    return Collections.emptyList();
  }

  @NonNull
  @Override
  public List<ViewManager> createViewManagers(@NonNull ReactApplicationContext reactContext) {
    return Collections.emptyList();
  }
}
```

**MainApplication.java**
```
package com.giuseppe.ravida.exponativeexamplev331;

import android.app.Application;
import android.content.res.Configuration;
import androidx.annotation.NonNull;

import com.facebook.react.PackageList;
import com.facebook.react.ReactApplication;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.config.ReactFeatureFlags;
import com.facebook.react.defaults.DefaultNewArchitectureEntryPoint;
import com.facebook.react.defaults.DefaultReactNativeHost;
import com.facebook.soloader.SoLoader;

import expo.modules.ApplicationLifecycleDispatcher;
import expo.modules.ReactNativeHostWrapper;

import java.util.List;

public class MainApplication extends Application implements ReactApplication {

  private final ReactNativeHost mReactNativeHost =
    new ReactNativeHostWrapper(this, new DefaultReactNativeHost(this) {
      @Override
      public boolean getUseDeveloperSupport() {
        return BuildConfig.DEBUG;
      }

      @Override
      protected List<ReactPackage> getPackages() {
        @SuppressWarnings("UnnecessaryLocalVariable")
        List<ReactPackage> packages = new PackageList(this).getPackages();
        // Packages that cannot be autolinked yet can be added manually here, for example:
        packages.add(new XyzFrameProcessorPluginPackage());
        return packages;
      }

      @Override
      protected String getJSMainModuleName() {
        return ".expo/.virtual-metro-entry";
      }

      @Override
      protected boolean isNewArchEnabled() {
        return BuildConfig.IS_NEW_ARCHITECTURE_ENABLED;
      }

      @Override
      protected Boolean isHermesEnabled() {
        return BuildConfig.IS_HERMES_ENABLED;
      }
  });

  @Override
  public ReactNativeHost getReactNativeHost() {
    return mReactNativeHost;
  }

  @Override
  public void onCreate() {
    super.onCreate();
    SoLoader.init(this, /* native exopackage */ false);
    if (!BuildConfig.REACT_NATIVE_UNSTABLE_USE_RUNTIME_SCHEDULER_ALWAYS) {
      ReactFeatureFlags.unstable_useRuntimeSchedulerAlways = false;
    }
    if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED) {
      // If you opted-in for the New Architecture, we load the native entry point for this app.
      DefaultNewArchitectureEntryPoint.load();
    }
    ReactNativeFlipper.initializeFlipper(this, getReactNativeHost().getReactInstanceManager());
    ApplicationLifecycleDispatcher.onApplicationCreate(this);
  }

  @Override
  public void onConfigurationChanged(@NonNull Configuration newConfig) {
    super.onConfigurationChanged(newConfig);
    ApplicationLifecycleDispatcher.onConfigurationChanged(this, newConfig);
  }
}

```

### Reproduceable Code

```tsx
import { StatusBar } from 'expo-status-bar';
import { Button, StyleSheet, Text, TouchableHighlight, View } from 'react-native';
import {  useCameraDevice, Camera, useFrameProcessor, runAsync, runAtTargetFps } from "react-native-vision-camera";
import NoCameraDeviceError from './NoCameraDeviceError';
import { useEffect, useState } from 'react';
import { OCRFrame, xyz } from './myFrameProcessor';

export default function App() {
	const [ showCamera, setShowCamera ] = useState(false);

	const device = useCameraDevice('back');

	useEffect(() => {
		const requestPermissions = async () => await Camera.requestCameraPermission();

		requestPermissions();
	}, []);

	const frameProcessor = useFrameProcessor((frame) => {
		'worklet'
		
		runAsync(frame, () => {
		 	'worklet'

		 	const response = xyz(frame);
		 	console.log(response, Date.now());	
		});
	}, []);

	if (device == null) return <NoCameraDeviceError />

	return (
		<>
			<StatusBar style={ showCamera ? 'light' : 'dark' } />

			{
				showCamera 
				&& 
				<>
					<Camera 
						device={device}
						style={StyleSheet.absoluteFill}
						isActive={showCamera}
						frameProcessor={frameProcessor}
						pixelFormat='yuv'
					/>

					<TouchableHighlight 
						onPress={() => setShowCamera(false)} 
						style={{ 
							position: "absolute", 
							bottom: 0, 
							width: "100%", 
							height: "10%",
							justifyContent: "center",
							alignItems: "center",
							backgroundColor: "rgba(255,255,255,0.5)"
						}}
					>
						<Text style={{ fontSize: 20, fontWeight: "bold" }}>Close</Text>
					</TouchableHighlight>
				</>
			}

			{
				!showCamera
				&&
				<View style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
					<Button title="📸 Open Camera" onPress={() => setShowCamera(true)} />
				</View>
			}
		</>		
	);
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#fff',
		alignItems: 'center',
		justifyContent: 'center',
	},
});
```


### Relevant log output

```shell
java.lang.reflect.InvocationTargetException
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at expo.modules.ReactNativeHostWrapperBase.invokeDelegateMethod$expo_debug(ReactNativeHostWrapperBase.kt:105)
                                                                                                    	at expo.modules.ReactNativeHostWrapperBase.getPackages(ReactNativeHostWrapperBase.kt:77)
                                                                                                    	at com.facebook.react.ReactNativeHost.createReactInstanceManager(ReactNativeHost.java:86)
                                                                                                    	at expo.modules.ReactNativeHostWrapperBase.createReactInstanceManager(ReactNativeHostWrapperBase.kt:32)
                                                                                                    	at com.facebook.react.ReactNativeHost.getReactInstanceManager(ReactNativeHost.java:42)
                                                                                                    	at expo.modules.devlauncher.launcher.DevLauncherNetworkInterceptor.shouldEmitEvents(DevLauncherNetworkInterceptor.kt:45)
                                                                                                    	at expo.modules.devlauncher.launcher.DevLauncherNetworkInterceptor.dispatch(DevLauncherNetworkInterceptor.kt:56)
                                                                                                    	at expo.modules.kotlin.devtools.ExpoRequestCdpInterceptor$dispatchEvent$1.invokeSuspend(ExpoRequestCdpInterceptor.kt:35)
                                                                                                    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                                                                                                    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
                                                                                                    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
                                                                                                    	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@1538f4c, Dispatchers.Default]
                                                                                                    Caused by: java.lang.AssertionError: Tried to add a Frame Processor Plugin with a name that already exists! Either choose unique names, or remove the unused plugin. Name: xyz
                                                                                                    	at com.mrousavy.camera.frameprocessor.FrameProcessorPluginRegistry.addFrameProcessorPlugin(FrameProcessorPluginRegistry.java:19)
                                                                                                    	at com.giuseppe.ravida.exponativeexamplev331.XyzFrameProcessorPluginPackage.<init>(XyzFrameProcessorPluginPackage.java:15)
                                                                                                    	at com.giuseppe.ravida.exponativeexamplev331.MainApplication$1.getPackages(MainApplication.java:35)
                                                                                                    	... 15 more
```


### Camera Device

```json
{
  "sensorOrientation": "landscape-right",
  "hardwareLevel": "full",
  "maxZoom": 8,
  "minZoom": 1,
  "supportsLowLightBoost": false,
  "neutralZoom": 1,
  "physicalDevices": [
    "wide-angle-camera"
  ],
  "supportsFocus": true,
  "supportsRawCapture": true,
  "isMultiCam": false,
  "name": "BACK (0)",
  "hasFlash": true,
  "hasTorch": true,
  "position": "back",
  "id": "0"
}
```


### Device

Xiaomi Redmi Note 7 - Android 10

### VisionCamera Version

3.4.1

### Can you reproduce this issue in the VisionCamera Example app?

I didn't try (⚠️ your issue might get ignored & closed if you don't try this)

### Additional information

- [X] I am using Expo
- [X] I have enabled Frame Processors (react-native-worklets-core)
- [X] I have read the [Troubleshooting Guide](https://react-native-vision-camera.com/docs/guides/troubleshooting)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/mrousavy/react-native-vision-camera/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched for [similar issues in this repository](https://github.com/mrousavy/react-native-vision-camera/issues) and found none.

--- a/docs/docs/guides/FRAME_PROCESSOR_CREATE_PLUGIN_ANDROID.mdx
+++ b/docs/docs/guides/FRAME_PROCESSOR_CREATE_PLUGIN_ANDROID.mdx
@@ -88,7 +88,7 @@ import com.mrousavy.camera.frameprocessor.FrameProcessorPluginRegistry;
 
 public class FaceDetectorFrameProcessorPluginPackage implements ReactPackage {
   // highlight-start
-  FaceDetectorFrameProcessorPluginPackage() {
+  static {
     FrameProcessorPluginRegistry.addFrameProcessorPlugin("detectFaces", options -> new FaceDetectorFrameProcessorPlugin(options));
   }
   // highlight-end
@@ -158,9 +158,11 @@ import com.mrousavy.camera.frameprocessor.FrameProcessorPlugin
 
 class FaceDetectorFrameProcessorPluginPackage : ReactPackage {
   // highlight-start
-  init {
-    FrameProcessorPluginRegistry.addFrameProcessorPlugin("detectFaces") { options ->
-      FaceDetectorFrameProcessorPlugin(options)
+  companion object {
+    init {
+      FrameProcessorPluginRegistry.addFrameProcessorPlugin("detectFaces") { options ->
+        FaceDetectorFrameProcessorPlugin(options)
+      }
     }
   }
   // highlight-end

--- a/package/example/android/app/src/main/java/com/mrousavy/camera/example/MainApplication.java
+++ b/package/example/android/app/src/main/java/com/mrousavy/camera/example/MainApplication.java
@@ -18,6 +18,11 @@ import com.mrousavy.camera.frameprocessor.FrameProcessorPlugin;
 import com.mrousavy.camera.frameprocessor.FrameProcessorPluginRegistry;
 
 public class MainApplication extends Application implements ReactApplication {
+  // Register the Frame Processor Plugins for our app
+  static {
+    FrameProcessorPluginRegistry.addFrameProcessorPlugin("example_plugin", options -> new ExampleFrameProcessorPlugin(options));
+    FrameProcessorPluginRegistry.addFrameProcessorPlugin("example_kotlin_swift_plugin", options -> new ExampleKotlinFrameProcessorPlugin(options));
+  }
 
   private final ReactNativeHost mReactNativeHost =
       new DefaultReactNativeHost(this) {
@@ -64,8 +69,5 @@ public class MainApplication extends Application implements ReactApplication {
       // If you opted-in for the New Architecture, we load the native entry point for this app.
       DefaultNewArchitectureEntryPoint.load();
     }
-
-    FrameProcessorPluginRegistry.addFrameProcessorPlugin("example_plugin", options -> new ExampleFrameProcessorPlugin(options));
-    FrameProcessorPluginRegistry.addFrameProcessorPlugin("example_kotlin_swift_plugin", options -> new ExampleKotlinFrameProcessorPlugin(options));
   }
 }

